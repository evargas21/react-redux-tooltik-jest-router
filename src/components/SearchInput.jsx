// @packages
import * as ProtoTypes from 'prop-types';

const SearchInput = ({ name, onChange, placeholder }) => {
    return (
        <div className="search-bar">
            <input
                data-testid="test-input-search"
                type="text"
                placeholder={placeholder}
                onChange={onChange}
                name={name}
            />
            <div>
                <button><span className="material-icons">search</span></button>
            </div>
        </div>
    );
};

SearchInput.defaulttProps = {
    placeholder: ''
};

SearchInput.propTypes = {
    name: ProtoTypes.string.isRequired,
    onChange: ProtoTypes.func.isRequired,
    placeholder: ProtoTypes.string
};

export default SearchInput;
