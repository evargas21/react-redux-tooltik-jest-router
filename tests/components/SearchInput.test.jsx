// @packages
import { render, screen } from '@testing-library/react';

// @scripts
import SearchInput from '../../src/components/SearchInput';

describe('Test in <SearchInput />', () => {
    const value = 'test';

    test('must match snapshot', () => {
        const { container } = render(<SearchInput  name={name} onChange={jest.fn()}/>)
        expect(container).toMatchSnapshot();
    });

    test('must display the input with the indicated value and placeholder', () => {
        render(<SearchInput name={value} onChange={jest.fn()} placeholder={value} />);
        const { name, placeholder } = screen.getByTestId('test-input-search');
        expect(name).toBe(value);
        expect(placeholder).toBe(value);
    });
});
