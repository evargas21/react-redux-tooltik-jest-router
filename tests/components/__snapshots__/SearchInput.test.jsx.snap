// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchInput /> must match snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <input
      data-testid="test-input-search"
      name=""
      type="text"
    />
    <div>
      <button>
        <span
          class="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
</div>
`;
